src/java/org/apache/cassandra/index/sai/iterators/KeyRangeIntersectionIterator.java

```diff
@@ -249,14 +249,24 @@ private void updateStatistics(Statistics statistics, KeyRangeIterator range)
 
     private static class IntersectionStatistics extends KeyRangeIterator.Builder.Statistics
     {
+        private boolean empty = true;
+
         @Override
         public void update(KeyRangeIterator range)
         {
             // minimum of the intersection is the biggest minimum of individual iterators
             min = nullSafeMax(min, range.getMinimum());
             // maximum of the intersection is the smallest maximum of individual iterators
             max = nullSafeMin(max, range.getMaximum());
-            count += range.getCount();
+            if (empty)
+            {
+                empty = false;
+                count = range.getCount();
+            }
+            else
+            {
+                count = Math.min(count, range.getCount());
+            }
```

Comment:
Can this be `count = count == 0 ? range.getCount() : Math.min(count, range.getCount());
`?

Reply:
I don't think so.  Consider adding two iterators A and B.  A has count of 0, B has count of 1.

If we add A first, then B, then we will end up with a count of 1, since it's treating 0 as "uninitialized" which is not correct.

Reply:
Ah, I see.